qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,117 @@
+def get_hpt(test, params, session, hpt_size, get_cmd):
+    """Run this step to check HPT in guest"""
+    if params.get("mistype") != "negative":
+        get_hpt_value = session.cmd_output(get_cmd, timeout=60)
+        if int(get_hpt_value) != hpt_size:
+            test.fail("HPT expected results aren't consistent")
+
+
+def mem_sub_hpt(test, vm):
+    """Run this step to hot plug 1G memory to guest"""
+    error_context.context("hotplug 1G memory before resizing HPT")
+    mem_id = utils_misc.generate_random_string(8)
+    dimm_id = utils_misc.generate_random_string(8)
+    vm.monitor.cmd_qmp("object-add", {
+                           "qom-type": "memory-backend-ram",
+                           "id": "%s" % mem_id,
+                           'props': {'policy': 'default', 'size': 1073741824}})
+    vm.monitor.cmd_qmp("device_add", {
+                           "driver": "pc-dimm",
+                           "id": "%s" % dimm_id,
+                           "memdev": "%s" % mem_id})
+    output = str(vm.monitor.info("memory-devices"))
+    logging.debug("Output of hotplugged memory : '%s'" % output)
+    if not re.search(r"%s" % mem_id, output, re.I):
+        test.fail("Memory backend can't be found ")
+    if not re.search(r"%s" % dimm_id, output, re.I):
+        test.fail("Pc-dimm can't be found ")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU HPT miscellaneous test:
+    Scenario 1
+    1) Start a HPT guest on Power8 or Power9 host
+    2) Increase HPT
+    3) Reboot
+
+    Scenario 2
+    1) Start a HPT guest on Power8 or Power9 host
+    2) Reduce HPT
+    3) Reboot
+
+    Scenario 3
+    1) Hotplug 1G memory to the guest
+    2) Increase HPT
+    3) Decrease HPT
+
+    Scenario 4
+    1) Start a HPT guest on Power8 and Power9 host
+    2) Resize hpt consectively with different params
+    3) Check guest work well or not
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Start miscellaneous HPT test...", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=600)
+    get_cmd = "cat /sys/kernel/debug/powerpc/hpt_order"
+    st, get_hpt_def = session.cmd_status_output(get_cmd, timeout=30)
+    if st:
+        test.cancel("Can't find available HPT value so skip test")
+    if params.get("mistype") == "mem":
+        mem_sub_hpt(test, vm)
+    hpt_size = int(get_hpt_def)
+    logging.debug("Default hpt size : '%s'" % get_hpt_def)
+    increment = params.get("increment").split()
+    error_context.context("hpt changes according to increment",
+                          logging.info)
+    for increm in increment:
+        hpt_size = hpt_size + int(increm)
+        set_hpt(vm, test, hpt_size)
+        get_hpt(test, params, session, hpt_size, get_cmd)
+    if "reboot" in params.get("mistype"):
+        vm.reboot()
```

Comment:
-> `session = vm.reboot(session)` to avoid creating the session manually.

Reply:
Updated it already